feat: editTrigger prop for editable Paragraph

kachanovskiy opened the pull request:
<!--
First of all, thank you for your contribution! 😄

New feature please send a pull request to feature branch, and rest to master branch.
Pull requests will be merged after one of the collaborators approve.
Please makes sure that these forms are filled before submitting your pull request, thank you!
-->

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [X] Site / documentation update
- [X] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [X] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

<!--
1. Describe the source of requirement, like related issue link.
-->

### 💡 Background and solution
in order to be able to use Paragraph as true inline edit (with no extra elements), it may be useful to hide the icon and use the value itself as edit trigger.

<!--
1. Describe the problem and the scenario.
2. GIF or snapshot should be provided if includes UI/interactive modification.
3. How to fix the problem, and list final API implementation and usage sample if that is a new feature.
-->

### 📝 Changelog

<!--
Describe changes from the user side, and list all potential break changes or other risks.
--->

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | for editable Paragraph, edit can be triggered by clicking the icon or the value itself (or both) |
| 🇨🇳 Chinese | 对于可编辑段落，可以通过点击图标或值本身（或两者都有）来触发进入编辑状态。 |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

kachanovskiy committed:
feat: editTrigger prop for editable Paragraph

afc163 commented on components/typography/index.zh-CN.md:
```diff
@@ -96,6 +96,8 @@ cover: https://gw.alipayobjects.com/zos/alicdn/GOM1KQ24O/Typography.svg
       onChange: function(string),
       onCancel: function,
       onEnd: function,
+      editTrigger: 'icon' | 'value' | 'both',
```
```suggestion
      triggerType: 'icon' | 'text' | 'both',
```

afc163 commented on components/typography/index.zh-CN.md:
```diff
@@ -111,6 +113,8 @@ cover: https://gw.alipayobjects.com/zos/alicdn/GOM1KQ24O/Typography.svg
 | onCancel | 按 ESC 退出编辑状态时触发 | function | - |  |
 | onEnd | 按 ENTER 结束编辑状态时触发 | function | - |  |
+| editTrigger | 编辑模式触发器 - 图标、值（文本）或两者皆有（在有“值”的情况下图标会被隐藏） | `icon`\|`value`\|`both` | `icon` |  |
+| alwaysShowEditIcon | 当 editTrigger = "value" 时被使用 - 图标默认是隐藏的，但可以将它显示出来 | boolean | false |  |
```
是否需要这个属性？我建议先不加

kachanovskiy committed:
feat: editTrigger - addressed review comments (changed prop name/value, removed alwaysShowIcon)

afc163 commented on components/typography/index.zh-CN.md:
```diff
@@ -96,6 +96,7 @@ cover: https://gw.alipayobjects.com/zos/alicdn/GOM1KQ24O/Typography.svg
       onChange: function(string),
       onCancel: function,
       onEnd: function,
+      triggerType: 'icon' | 'text' | 'both',
```
```suggestion
      triggerType: ('icon' | 'text')[],
```

afc163 replied:
`both` => `['icon', 'text']`

afc163 commented:
Conflicting

afc163 commented:
ping @kachanovskiy

kachanovskiy committed:
feat: triggerType (was editTrigger) - addressed more review comments, updated documentation

kachanovskiy committed:
feat: triggerType (was editTrigger) - addressed more review comments, added missing empty lines in doc

kachanovskiy committed:
feat: triggerType (was editTrigger) - merge conflict

kachanovskiy committed:
feat: triggerType (was editTrigger) - merge conflict

afc163 commented:
CI is broken, could you check it?

kachanovskiy commented:
I am sorry, I am unable to identify what's wrong - locally it builds just fine, and CI run log does not have enough of data I could have used to look into this. Could you please assist?

one thing i used to have at some point (and which then has gone) was errors due to postcss 8 required but not present...

afc163 commented:
```
npm test -- components/typography -u
git commit -am 'chore: update snapshot'
git push origin skachano_paragraph_editable_edit_trigger
```

kachanovskiy committed:
chore: update snapshot

afc163 commented:
<img width="679" alt="图片" src="https://user-images.githubusercontent.com/507615/130548269-c503ce1e-a4e0-4143-b961-b5b1242e6628.png">

Test coverage is down.

afc163 commented:
ping @kachanovskiy

kachanovskiy commented:
apologies, this would have to wait until i am back next thursday, i am away right now…sorry about this.
________________________________
From: afc163 ***@***.***>
Sent: Saturday, September 4, 2021 12:39:08 PM
To: ant-design/ant-design ***@***.***>
Cc: Sergey Kachanovskiy ***@***.***>; Mention ***@***.***>
Subject: Re: [ant-design/ant-design] feat: editTrigger prop for editable Paragraph (#31537)


ping @kachanovskiy<https://github.com/kachanovskiy>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/ant-design/ant-design/pull/31537#issuecomment-912951271>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADAQCQDIM4I2JETTGMSBLVDUAHZMZANCNFSM5BA3NJVA>.

xrkffgg committed:
branch: merge feature into master (#32050)

xrkffgg committed:
chore: down node-fetch version

3.x ESM only。scripts 中几个校验不支持。使用低版本。

xrkffgg committed:
fix: Upload icon error background (#32051)

lironhl committed:
fix: List rowKey prop type (#32033)

afc163 committed:
fix: Alert message could be nullable (#32057)

* fix: Alert message could be nullable

close #32047

* update snapshot

zombieJ committed:
fix: Less Maximum call stack size exceeded error (#32063)

* chore: create additional entry

* fix: rm recv call

* chore: Add default.less as index.less

* chore: update entry

* fix: row should also translate

* chore: rename index-default to index and add index-pure instead

* fix: missing transfer customize styl

zombieJ committed:
chore: 4.17.0-alpha.2 changelog (#32068)

btea committed:
fix: set the tree-switcher-leaf-line position (#32075)

drizzlesconsin committed:
fix(image): small size hide preview text (#29900)

* fix(image): small size hide preview text

* fix: image preview text ellipsis

* fix: reset test

afc163 committed:
chore: improve icon search experience (#32098)

* chore: improve icon search experience

close #32097

* chore: search result empty data

PrimulaX committed:
feat: Georgian translation (#32106)

* parent 18cd40167857648ff48f6f8db229e5fea7608922
author Primlx <primulax@live.com> 1628778277 +0400
committer primulax <primulax@live.com> 1631181358 +0400

Add files via upload

add georgian translation

Create ka_GE.tsx

Create ka_GE

Rename ka_GE to ka_GE.tsx

Create ka_GE.tsx

Update ka_GE.tsx

Update ka_GE.tsx

Create ka_GE.tsx

update rc-picker version

Create ka_GE.tsx

Create ka_GE

Rename ka_GE to ka_GE.tsx

Update ka_GE.tsx

Update ka_GE.tsx

Create ka_GE.tsx

update rc-picker version

* Update package.json

Co-authored-by: Amumu <yoyo837@hotmail.com>

* test case

* add language option in markdown

* test

* Update components/calendar/locale/ka_GE.tsx

Co-authored-by: Amumu <yoyo837@hotmail.com>

* Update components/calendar/locale/ka_GE.tsx

Co-authored-by: Amumu <yoyo837@hotmail.com>

Co-authored-by: Amumu <yoyo837@hotmail.com>

in-vane committed:
docs: Update research-message-and-feedback.zh-CN.md (#32112)

thommesZip committed:
fix: Use latest rc-picker version with defaultPickerValue/showTime fix (#32107)

* update rc-picker dependency to 2.5.17

* add test: DatePicker.RangePicker with defaultPickerValue and showTime

zombieJ committed:
fix: missing `default.css` (#32116)

* chore: Add CI for compile

* chore: Add missing compile file

arifszn committed:
docs: fix typo (#32121)

JounQin committed:
fix: preserve table pagination className (#32131)

close #32130

JounQin committed:
fix: use well typed PaginationLocale from rc-pagination (#32128)

yoyo837 committed:
docs: fix typo

mochengdx committed:
fix: useMessage() should use getPopupContainer from <ConfigProvider />… (#31939)

* fix:useMessage() should use getPopupContainer from <ConfigProvider /> #31841

* fix: useMessage() should use getPopupContainer from <ConfigProvider /> #31841

* add test case for useMessage hook

* restore some changes

* try to restore files agains

* restore and keep the same as upstream master

* restore files

* restore some definition

JounQin committed:
feat: inherit rc-pagination props, more customizable (#32132)

* feat: inherit rc-pagination props, more customizable

close #32128, #32129

* chore: CustomSelect.Option is required by rc-pagination

* fix: LocaleReceivier types issue

* fix: lint issue, pageSize must be presented

jaredleechn committed:
chore: add Button.shape 'default'  definition (#32004)

* feat: add Button.shape 'default'

* fix: ignore shape when falsy value

* fix: generate className only when shape match

* fix: remove necessary .ant-btn-default and update snapshot

zombieJ committed:
fix: Cascader missing props (#32143)

* fix: Cascader deps

* docs: Update using new API

* test: Test case cover

* chore: compressed skip dup check

JounQin committed:
chore: remove redundant space after `should` (#32141)

dependabot[bot] committed:
chore(deps-dev): bump eslint-plugin-unicorn from 35.0.0 to 36.0.0 (#32149)

Bumps [eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn) from 35.0.0 to 36.0.0.
- [Release notes](https://github.com/sindresorhus/eslint-plugin-unicorn/releases)
- [Commits](https://github.com/sindresorhus/eslint-plugin-unicorn/compare/v35.0.0...v36.0.0)

---
updated-dependencies:
- dependency-name: eslint-plugin-unicorn
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

afc163 committed:
chore: fix modal demo in small screen

Dreamerryao committed:
refactor(dropdown): add dropdownButtonType export and refactor defaultProps (#31957)

vagusX committed:
docs: add changelog 4.17.0-alpha.3 (#32147)

zombieJ committed:
chore: Not preventEvent btn loading (#32158)

* chore: Not preventEvent btn loading

* test: Update snapshot

zombieJ committed:
fix: Col with `flex` exceed its limit (#32160)

* fix: Col flex exceed its value

* test: Update snapshot

Rain120 committed:
docs: add DatePicker format demo(Week start ~ end) (#32078)

* feat: doc of week format in date-picker

* feat: doc of week format in date-picker

* fix: set the tree-switcher-leaf-line position (#32075)

* fix(image): small size hide preview text (#29900)

* fix(image): small size hide preview text

* fix: image preview text ellipsis

* fix: reset test

* chore: improve icon search experience (#32098)

* chore: improve icon search experience

close #32097

* chore: search result empty data

* feat: Georgian translation (#32106)

* parent 18cd40167857648ff48f6f8db229e5fea7608922
author Primlx <primulax@live.com> 1628778277 +0400
committer primulax <primulax@live.com> 1631181358 +0400

Add files via upload

add georgian translation

Create ka_GE.tsx

Create ka_GE

Rename ka_GE to ka_GE.tsx

Create ka_GE.tsx

Update ka_GE.tsx

Update ka_GE.tsx

Create ka_GE.tsx

update rc-picker version

Create ka_GE.tsx

Create ka_GE

Rename ka_GE to ka_GE.tsx

Update ka_GE.tsx

Update ka_GE.tsx

Create ka_GE.tsx

update rc-picker version

* Update package.json

Co-authored-by: Amumu <yoyo837@hotmail.com>

* test case

* add language option in markdown

* test

* Update components/calendar/locale/ka_GE.tsx

Co-authored-by: Amumu <yoyo837@hotmail.com>

* Update components/calendar/locale/ka_GE.tsx

Co-authored-by: Amumu <yoyo837@hotmail.com>

Co-authored-by: Amumu <yoyo837@hotmail.com>

* docs: Update research-message-and-feedback.zh-CN.md (#32112)

* fix: Use latest rc-picker version with defaultPickerValue/showTime fix (#32107)

* update rc-picker dependency to 2.5.17

* add test: DatePicker.RangePicker with defaultPickerValue and showTime

* fix: missing `default.css` (#32116)

* chore: Add CI for compile

* chore: Add missing compile file

* docs: fix typo (#32121)

* fix: preserve table pagination className (#32131)

close #32130

* fix: use well typed PaginationLocale from rc-pagination (#32128)

* docs: fix typo

* fix: useMessage() should use getPopupContainer from <ConfigProvider />… (#31939)

* fix:useMessage() should use getPopupContainer from <ConfigProvider /> #31841

* fix: useMessage() should use getPopupContainer from <ConfigProvider /> #31841

* add test case for useMessage hook

* restore some changes

* try to restore files agains

* restore and keep the same as upstream master

* restore files

* restore some definition

* feat: inherit rc-pagination props, more customizable (#32132)

* feat: inherit rc-pagination props, more customizable

close #32128, #32129

* chore: CustomSelect.Option is required by rc-pagination

* fix: LocaleReceivier types issue

* fix: lint issue, pageSize must be presented

* chore: add Button.shape 'default'  definition (#32004)

* feat: add Button.shape 'default'

* fix: ignore shape when falsy value

* fix: generate className only when shape match

* fix: remove necessary .ant-btn-default and update snapshot

* fix: Cascader missing props (#32143)

* fix: Cascader deps

* docs: Update using new API

* test: Test case cover

* chore: compressed skip dup check

* chore: remove redundant space after `should` (#32141)

* chore(deps-dev): bump eslint-plugin-unicorn from 35.0.0 to 36.0.0 (#32149)

Bumps [eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn) from 35.0.0 to 36.0.0.
- [Release notes](https://github.com/sindresorhus/eslint-plugin-unicorn/releases)
- [Commits](https://github.com/sindresorhus/eslint-plugin-unicorn/compare/v35.0.0...v36.0.0)

---
updated-dependencies:
- dependency-name: eslint-plugin-unicorn
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* chore: fix modal demo in small screen

* refactor(dropdown): add dropdownButtonType export and refactor defaultProps (#31957)

Co-authored-by: btea <2356281422@qq.com>
Co-authored-by: drizzlesconsin <69514654+drizzlesconsin@users.noreply.github.com>
Co-authored-by: afc163 <afc163@gmail.com>
Co-authored-by: Primlx <primulax@live.com>
Co-authored-by: Amumu <yoyo837@hotmail.com>
Co-authored-by: in_vane <52909860+in-vane@users.noreply.github.com>
Co-authored-by: Thomas Zipner <thomas.zipner@gmail.com>
Co-authored-by: 二货机器人 <smith3816@gmail.com>
Co-authored-by: MD. Ariful Alam <swazan.arif@gmail.com>
Co-authored-by: JounQin <admin@1stg.me>
Co-authored-by: James <daoxingyue@live.cn>
Co-authored-by: 绯一 <jaredleechn@users.noreply.github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Zuobin Wang <49583235+Dreamerryao@users.noreply.github.com>

kachanovskiy committed:
feat: triggerType - unit test update to improve coverage

kachanovskiy committed:
Merge branch 'ant-design:master' into skachano_paragraph_editable_edit_trigger

kachanovskiy commented:
@afc163 seems like i did something wrong, could you please help me with this? thanks...

kachanovskiy commented:
ok seems like i did "fetch upstream" in a wrong branch, which has caused a merge of master into my branch in my fork.

am i getting it right that i can either:
1) do "git push --force" from my local copy where 8a46d6962005d26e5ffbfa84a8c1cfb369cd1dbd is still on top
2) or clone my repo from github, checkout my branch, do "git reset --hard HEAD~1" and then "git push --force" which is supposed to bring my repo back to where it was before my erroneous action?

kachanovskiy commented:
@afc163 i have removed merge commit from the erroneous upstream fetch from the branch in my fork, please advise what should i do now - would you re-open this PR, or should i create a new one from the same branch and go through all the steps again?

afc163 commented:
You can create a new PR from scratch.

xrkffgg commented:
Next can use https://github.com/ant-design/ant-design/issues/30441
